master/buildbot/worker/protocols/manager/msgpack.py

```diff
@@ -344,10 +355,38 @@ def get_message_result(self, msg):
         res1 = yield d
         return res1
 
+    @defer.inlineCallbacks
     def onConnect(self, request):
         if self.debug:
             log.msg(f"Client connecting: {request.peer}")
 
+        try:
+            value = request.headers['authorization']
+            username, password = decode_http_authorization_header(value)
+        except Exception as e:
+            raise ConnectionDeny(401, "Unauthorized") from e
```

Comment:
if there is no autentication header, then we should throw 401 Unauthorized

If the authentication header is malformated: 400 Bad request

Reply:
Fixed.